src/sumo_event.erl

```diff
@@ -41,6 +46,7 @@ dispatch(DocName, Event, Args) ->
       ok;
     EventManagers ->
       lists:foreach(fun(EventManager) ->
-        gen_event:notify(EventManager, {DocName, Event, Args})
+        EventId = make_ref(),
```

Comment:
This will not work… `EventId` must be preserved between linked instances of `pre_` and regular events